File: FSharp/FrgJmpCodility.fs

let rec solve(startVal, dVal, endVal, count) = 
    if startVal >= endVal then
        count
    else
        solve(startVal + dVal, dVal, endVal, count + 1)
     

let solution(x, y, d) =
    solve(x, d, y, 0)



[<EntryPoint>]
let main argv = 
    let result = solution(10,850000000,30)
    printfn "%A" result
    0 

File: TypeScript/FrgJmpCodility.ts
const solve = (startVal, dVal, endVal, count) => {
  if (startVal >= endVal) {
      return count;
  }
  else{
    return solve(startVal + dVal, dVal, endVal, count + 1);
  }
}

const solution = (x, y, d) => {
  return solve(x, d, y, 0);
}

console.log(solution(10,85,30))
